openhtf/util/data.py

```diff
@@ -126,10 +126,13 @@ def convert_to_base_types(obj, ignore_keys=tuple(), tuple_type=tuple):
   if hasattr(obj, '_asdict'):
     obj = obj._asdict()
   elif isinstance(obj, records.RecordClass):
-    obj = {attr: getattr(obj, attr)
-           for attr in type(obj).all_attribute_names
-           if (getattr(obj, attr) is not None or
-               attr in type(obj).required_attributes)}
+    temp = {}
+    for attr in type(obj).all_attribute_names:
+      if hasattr(obj, attr):
```

Comment:
i think it would be better to `getattr(obj, attr, None)` instead of doing the hasattr check